splicemachine/spliceengine
# DB-12051 Add support for additional constant folding

## Short Description
Support NOT, unary minus, conditional and binary relational operator constant folding


## Long Description
- **DB-12145** NOT: Support for constant folding added for null and non null operands

- **DB-12148** Conditional node: Support for constant folding added for null and non null operands. If the condition is constant but then or else isn't, remove the condition and return the correct underlying operand

- **DB-12146** Unary plus / minus: Support for constant folding added for nulls and non null number operand.

- **DB-12147** BinaryRelationalOperatorNode: Support added for null operands. Non null operand constant folding was already implemented.

## How to test
This can be tested via an explain on a simple select: 
```
explain select * from sysibm.sysdummy1 where 1 = -(1)
```
will return a plan without a table scan because `1 = -(1)` is evaluated to be false at compile time
The same principle will apply to `true = not false`, `true = case when true then true else false end`, etc.
A new IT was created and should be referred to for more details.
